src/main/java/com/openClassroomsProject/SafetyNetAlerts/service/IMedicalRecordService.java

```diff
@@ -0,0 +1,17 @@
+package com.openClassroomsProject.SafetyNetAlerts.service;
+
+import com.openClassroomsProject.SafetyNetAlerts.model.MedicalRecord;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+
+@Service
+public interface IMedicalRecordService {
+
+    Iterable<MedicalRecord> getMedicalRecords();
+
+    void addMedicalRecord(MedicalRecord medicalRecord);
+
+    Optional<MedicalRecord> updateAnExistingMedicalRecord(MedicalRecord medicalRecord);
+
+    boolean deleteAMedicalRecord(String firstName, String lastName);
```

Comment:
Just a quick advice, the best number of parameters in a function is 0, then if you can, don't hesitate to create a class which will encapsulate those properties. Thanks to this the day you need to update it, there will be no change on the fund. Expl:

`boolean deleteAMedicalRecord(MedicalRecord medicalRecord);`

Reply:
I understood, I did that because the unique identifier must be the lastname and the first name. But as you said last time, if at creation all the fields are mandatory then when deleting  the complete object can also be required I imagine.

Reply:
Or I have to create a class which has just lastname and fisrtname but... seem strange no ?

Reply:
Yeap it seems strange at first, but it can save your time later :D

Reply:
Indeed, creating a new object is better than injecting a lot of parameters

Reply:
Roger